oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/authorization/OidcConfigurer.java

```diff
@@ -63,11 +65,23 @@ public OidcConfigurer clientRegistrationEndpoint(Customizer<OidcClientRegistrati
 		return this;
 	}
 
+	/**
+	 * Configures the OAuth 2.0 Protected Resource UserInfo Endpoint.
+	 *
+	 * @param userInfoEndpointCustomizer the {@link Customizer} providing access to the {@link OidcUserInfoEndpointConfigurer}
+	 * @return the {@link OidcConfigurer} for further configuration
+	 */
+	public OidcConfigurer userInfoEndpoint(Customizer<OidcUserInfoEndpointConfigurer> userInfoEndpointCustomizer) {
+		userInfoEndpointCustomizer.customize(this.userInfoEndpointConfigurer);
+		return this;
+	}
+
 	@Override
 	<B extends HttpSecurityBuilder<B>> void init(B builder) {
 		if (this.clientRegistrationEndpointConfigurer != null) {
 			this.clientRegistrationEndpointConfigurer.init(builder);
 		}
+		this.userInfoEndpointConfigurer.init(builder);
```

Comment:
This is strictly a matter of personal preference, but how do you feel about moving this to the top of the function to highlight the fact that it's always contained in the request matchers (as opposed to the optional `clientRegistrationEndpointConfigurer`)? It can help make the `requestMatchers.size() > 1` condition a little more obvious.